libs/ml/include/ml/serializers/ml_types.hpp

```diff
@@ -1896,6 +1907,45 @@ struct MapSerializer<ml::OpMaxPool1DSaveableParams<TensorType>, D>
   }
 };
 
+/**
+ * serializer for MaxPool saveable params
+ * @tparam TensorType
+ */
+template <typename TensorType, typename D>
+struct MapSerializer<ml::OpMaxPoolSaveableParams<TensorType>, D>
+{
+  using Type       = ml::OpMaxPoolSaveableParams<TensorType>;
+  using DriverType = D;
+
+  static uint8_t const BASE_OPS    = 1;
+  static uint8_t const OP_CODE     = 2;
+  static uint8_t const KERNEL_SIZE = 3;
+  static uint8_t const STRIDE_SIZE = 4;
```

Comment:
If you make the change I suggested before you might also need to serialise something to indicate whether the shared_ptr<Ops> should be dynamically cast to 1D or 2D